src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,76 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int NUMBER_OF_BOUND = 6;
+    private static final int DEFAULT_SIZE = 10;
+    private static final String DEFAULT_COLOR = "WHITE";
+    private static final FigureSupplier figureSupplier = new FigureSupplier();
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int indexOfColor = random.nextInt(NUMBER_OF_BOUND);
+        Figure randomFigure;
+        switch (indexOfColor) {
+            case 1: {
+                randomFigure = createRandomSquare(colorSupplier);
+                break;
+            }
+            case 2: {
+                randomFigure = createRandomRectangle(colorSupplier);
+                break;
+            }
+            case 3: {
+                randomFigure = createRandomRightTriangle(colorSupplier);
+                break;
+            }
+            case 4: {
+                randomFigure = createRandomCircle(colorSupplier);
+                break;
+            }
+            case 5: {
+                randomFigure = createRandomIsoscelesTrapezoid(colorSupplier);
+                break;
+            }
+            default: {
+                return figureSupplier.getDefaultFigure();
+            }
+        }
+        return randomFigure;
+    }
```

Comment:
random.nextInt(NUMBER_OF_BOUND) will give us value from 0 to 5 (when NUMBER_OF_BOUND == 6).
this means that in about 17% of cases it will be 0 and as a result method will produce default figure
We need to start our cases with 0 and return NUMBER_OF_BOUND value back to 5.
With such changes default value will be used only if we messed up with code after adding new figures
```suggestion
        int indexOfColor = random.nextInt(NUMBER_OF_BOUND);
        Figure randomFigure;
        switch (indexOfColor) {
            case 0: {
                randomFigure = createRandomSquare(colorSupplier);
                break;
            }
            case 1: {
                randomFigure = createRandomRectangle(colorSupplier);
                break;
            }
            case 2: {
                randomFigure = createRandomRightTriangle(colorSupplier);
                break;
            }
            case 3: {
                randomFigure = createRandomCircle(colorSupplier);
                break;
            }
            case 4: {
                randomFigure = createRandomIsoscelesTrapezoid(colorSupplier);
                break;
            }
            default: {
                return figureSupplier.getDefaultFigure();
            }
        }
        return randomFigure;
    }
```